foundry/gui/orb.py

```diff
@@ -0,0 +1,362 @@
+        internal_layout = QBoxLayout(QBoxLayout.TopToBottom)
+        group.setLayout(internal_layout)
+
+        self._move_touch_to_timer = View._create_checkbox(
+            _UIStrings.MOVE_TOUCH_TO_TIMER.value, self._on_move_touch_to_timer, internal_layout
+        )
+
+        self._move_timer_to_exit = View._create_checkbox(
+            _UIStrings.MOVE_TIMER_TO_EXIT.value, self._on_move_timer_to_exit, internal_layout
+        )
+
+        external_layout.addWidget(group)
+        return external_layout
+
+    def _create_button_options_layout(self) -> QDialogButtonBox:
+        """Creates layout for the OK/CANCEL buttons."""
+
+        button_box = QDialogButtonBox()
+        button_box.addButton(QDialogButtonBox.Ok).clicked.connect(self._on_ok)
+        button_box.addButton(QDialogButtonBox.Cancel).clicked.connect(self._on_cancel)
+
+        return button_box
+
+    @staticmethod
+    def _create_checkbox(title: str, function, layout: QBoxLayout) -> QCheckBox:
+        """Creates a checkbox with an stateChange callback and adds it to the specified layout"""
+
+        checkbox = QCheckBox(f"{title}")
+        checkbox.stateChanged.connect(function)
+        layout.addWidget(checkbox)
+        return checkbox
+
+    def render(self):
+        """Updates the UI with the current state values.
+
+        This function is the subscriber to the store so that whenever there is
+        a state change in the system, this render function is called
+        automatically and the new state is rendered on screen.
+        """
+
+        state = self.store.get_state()
+
+        View._render_checkbox(self._move_touch_to_timer, state.move_touch_to_timer)
+        View._render_checkbox(self._move_timer_to_exit, state.move_timer_to_exit)
+        View._render_checkbox(self._touch_game_timer_stops, state.touch_game_timer_stops)
+
+        self._invalid_rom_warning.setVisible(View._all_areas_valid(state) is False)
+
+    @staticmethod
+    def _render_checkbox(checkbox: QCheckBox, value: bool):
+        """Render a checkbox value."""
+
+        checkbox.setDisabled(value is None)
+        checkbox.setChecked(False if value is None else value)
+
+    @staticmethod
+    def _all_areas_valid(state: State) -> bool:
+        """Checks to make sure all state values are valid."""
+        return (
+            state.move_touch_to_timer is not None
+            and state.move_timer_to_exit is not None
+            and state.touch_game_timer_stops is not None
+        )
+
+    def _on_ok(self):
+        """Process UI press of OK button."""
+        self.rom_interface.write_state(self.store.get_state())
+        self.done(QDialogButtonBox.Ok)
+
+    def _on_cancel(self):
+        """Process UI press of CANCEL button."""
+        self.done(QDialogButtonBox.Cancel)
+
+    def _on_move_touch_to_timer(self):
+        self.store.dispatch(Action(Actions.MOVE_TOUCH_TO_TIMER, self._move_touch_to_timer.isChecked()))
```

Comment:
The Action Enum is not a string.  I believe you can make Actions inherit from str and this will be resolved.

Reply:
I looked into this and it didn't seem as clean as just using .value.  I used .value in the player lives window so this will be consistent at least.